lib/main_publiccloud.pm

```diff
@@ -58,7 +58,10 @@ sub load_maintenance_publiccloud_tests {
             loadtest "publiccloud/az_l8s_nvme" if (get_var('PUBLIC_CLOUD_INSTANCE_TYPE') =~ 'Standard_L(8|16|32|64)s_v2');
         } elsif (get_var('PUBLIC_CLOUD_AZURE_NFS_TEST')) {
             loadtest("publiccloud/azure_nfs", run_args => $args);
+        } elsif (check_var('PUBLIC_CLOUD_NVIDIA', 1)) {
+            loadtest "publiccloud/nvidia", run_args => $args if (check_var('PUBLIC_CLOUD_PROVIDER', 'GCE') && is_sle('15-SP4+'));
```

Comment:
so if you will define variable PUBLIC_CLOUD_NVIDIA but will set PUBLIC_CLOUD_PROVIDER=Azure it will boot create instance and than simply shutdown ? I would suggest to move this checks into module and fail test there

```suggestion
            loadtest "publiccloud/nvidia", run_args => $args;
```

Reply:
No. First test is not designed or tested with Azure. And for now i think it doesnt matter (but maybe i will need to confirm). The think is that no all providers provide the GPU. The `is_sle('15-SP4+')` is required as the driver is not supported in older versions.

Reply:
yes I know that it is not designed for Azure ( or Amazon )  . and main point of this `if` to handle the case when someone mistakenly try to run this test against Azure (or Amazon ) if I get it right . And my point here that I don't like how you handle such case . In such case you will silently skip one module and that's it . I think better to : 
1. Not do such checks in main_publiccloud.pm
2. Move checks **inside** nvidia.pm
3. not just skip but die

Reply:
After offline discussion, i update the code with the following approach. Make the test die during the scheduling. With this we prevent any execution with the wrong requirements. http://aquarius.suse.cz/tests/13481

![image](https://user-images.githubusercontent.com/330547/197321899-87d50351-7fde-4e29-af11-b34ed15e4738.png)

Reply:
*ready for review*